src/ga/ga_scaling.rs

```diff
@@ -1,11 +1,107 @@
-// TODO: COPYRIGHT, USE & AUTHORS
+// Copyright 2016 Revolution Solid & Contributors.
+// author(s): sysnett, carlos-lopez-garces
+// rust-monster is licensed under an MIT License.
+
+//! GA Scaling Schemes
+//!
+//! Scales the raw score of a Population's individuals.
+
 use super::ga_core::GASolution;
+use super::ga_population::GAPopulation;
 
 /// Scaling Scheme Trait
-///
-/// Interface to a Scaling Scheme. Goes from 'raw' objective values
-/// to fitness values.
+/// 
+/// Embeeded in the population, scales the values of raw score in a
+/// GASolution to set their fitness score
 pub trait GAScaling<T: GASolution>
 {
-    fn evaluate(pop : &mut GAPopluation<T>);
+    fn evaluate(&self, pop: &mut GAPopulation<T>);
+}
+
+/// No Scaling - Raw and Scaled are the same
+pub struct GANoScaling;
+
+impl<T: GASolution> GAScaling<T> for GANoScaling
+{
+    fn evaluate(&self, pop: &mut GAPopulation<T>)
+    {
+        // TODO: This is why we need iterators :(
+        let pop_vec = pop.population();
+        for ref mut ind in pop_vec
+        {
+            let f = ind.score();
+           ind.set_fitness(f); 
+        }
+    }
+}
+
+/// Linear Scaling
+/// Uses a simple ```a*fitness + b``` scaling.
+/// ```a``` and ```b``` are the intersect of the linear function and are calculated
+/// based on Goldberg's book implementation
+pub struct GALinearScaling
+{
+    multiplier: f32
+}
+
+#[allow(unused_variables)]
+const GA_LINEAR_SCALING_MULTIPLIER : f32 = 2.0;
```

Comment:
I agree with all you said, the only use case I see (and why I added the const) is:

`
let myScalingThingy = GALinearScaling::new(GA_LINEAR_SCALING_MULTIPLIER);
`

So clients of the API have a good reference to pass instead of just guessing (and maybe passing a 1, dividing among 0 and getting random errors they can't debug).